### PR TITLE
API changes to requestedSpace()

### PR DESCRIPTION
**Pull request by jtlan**
Recreated version of #1933 with the correct branch name.

Based off of **develop** for ease of testing. Once QE gives the all-clear I'll merge in **apiRefactor**.

### CODE REVIEW

**Comment by crmorford** [comment]
woah, didn't even realize some legends weren't getting room to put everything before
develop:
screen shot 2015-05-04 at 6 33 08 pm

this branch:
screen shot 2015-05-04 at 6 33 16 pm

**Comment by crmorford** [comment]
kinda wanna call in design on this one to see which is better
screen shot 2015-05-04 at 6 34 49 pm
screen shot 2015-05-04 at 6 34 56 pm

**Comment by crmorford** [comment]
Overall: seems like legends are fighting for space less, plots are fighting for space more, and category axis is fighting for space much more. sound like what you expected?

**Comment by jtlan** [comment]
@crmorford That's roughly what I expected. `Axis.Category` actually had some bugs in its layout I addressed (didn't ask for more space even though it wanted some padding).

**Comment by jtlan** [comment]
@crmorford If you want to check this against the **apiRefactor** branch the quicktests should work there now too.

**Comment by crmorford** [comment]
btw, all the interactions feel so much better. 
back on topic: titleLegend_change breaks when you click the six plots label. looking into it. otherwise, looks good.

**Comment by crmorford** [comment]
if my test update commit passes travis, this is good to go.

**Comment by jtlan** [comment]
Putting a hold on this while I dig into the `Legend` issue.

**Comment by jtlan** [comment]
`Legend` now asks for more width if entry would be truncated:
screen shot 2015-05-05 at 5 35 36 pm

File: src/components/legend.ts [hunk 1]
```diff
@@ -141,16 +141,20 @@ export module Components {
...
-      var entries = this._scale.domain().slice();
-      entries.sort(this.sortFunction());
-      var entryLengths = Utils.Methods.populateMap(entries, measureEntry);
+      var entryNames = this._scale.domain().slice();
+      entryNames.sort(this.sortFunction());
+
+      var entryLengths: D3.Map<number> = d3.map();
+      var untruncatedEntryLengths: D3.Map<number> = d3.map();
+      entryNames.forEach((entryName) => {
+        var untruncatedEntryLength = (textHeight + this._measurer.measure(entryName).width + this._padding);
```
Comment (bluong) [inline src/components/legend.ts]: Parens here unnecessary

File: src/components/legend.ts [hunk 2]
```diff
@@ -167,26 +172,15 @@ export module Components {
...
-
-      var longestRowLength = Utils.Methods.max(rowLengths, 0);
 
-      var longestUntruncatedEntryLength = Utils.Methods.max<string, number>(this._scale.domain(), (d: string) =>
-                                            this._measurer.measure(d).width, 0);
-      longestUntruncatedEntryLength += estimatedLayout.textHeight + this._padding;
-      var desiredWidth = this._padding + Math.max(longestRowLength, longestUntruncatedEntryLength);
+      var untruncatedRowLengths = estimatedLayout.rows.map((row: string[]) => {
+        return d3.sum(row, (entry: string) => estimatedLayout.untruncatedEntryLengths.get(entry));
+      });
```
Comment (bluong) [inline src/components/legend.ts]: Is this workable?
```typescript
var untruncatedRowLengths = estimatedLayout.rows.map((row) => d3.sum(row, (entry) => estimatedLayout.untruncatedEntryLengths.get(entry))
```

Comment (jtlan) [inline src/components/legend.ts]: Probably too long to fit one one line. I think I can remove the inferred types though.

**Comment by jtlan** [comment]
@crmorford The only thing that changed should be `Legend`.

**Pull request merged**